lib/elixir_make/artefact.ex

```diff
@@ -187,22 +216,36 @@ defmodule ElixirMake.Artefact do
   @doc """
   Returns the url for the current target.
   """
-  def current_target_url(config, precompiler, nif_version) do
+  def current_target_url(config, precompiler, current_nif_version) do
     case precompiler.current_target() do
       {:ok, current_target} ->
-        available_urls = available_target_urls(config, precompiler)
-        target_at_nif_version = {current_target, nif_version}
+        nif_versions =
+          config[:make_precompiler_nif_versions] ||
+            [versions: []]
+
+        versions = nif_versions[:versions]
+        fallback_version = nif_versions[:fallback_version] || &find_nif_version/3
+        nif_version_to_use = fallback_version.(current_target, current_nif_version, versions)
```

Comment:
Sorry, we wqill need to change `find_nif_version/3` accordingly, to receive an extra argument and to return the `current_nif_version` instead of `:no_candiates`. :D

Reply:
Okay, it's done. ;)